frontend/components/AgentWallet/FundAgent/ConfirmTransfer.tsx

```diff
@@ -0,0 +1,238 @@
+      <WarningOutlined />
+    </Flex>
+
+    <Flex gap={12} vertical className="text-center">
+      <Title level={4} className="m-0">
+        Transfer Failed
+      </Title>
+      <Text className="text-neutral-tertiary">
+        Something went wrong with your transfer. Please try again or contact the
+        Olas community.
+      </Text>
+    </Flex>
+
+    <Flex gap={16} vertical className="text-center">
+      <Button onClick={onTryAgain} type="primary" block size="large">
+        Try Again
+      </Button>
+      <Link href={SUPPORT_URL} target="_blank" rel="noreferrer">
+        Join Olas Community Discord Server {UNICODE_SYMBOLS.EXTERNAL_LINK}
+      </Link>
+    </Flex>
+  </Flex>
+);
+
+type TokenAmountMap = {
+  [address: Address]: string; // amount (in wei/units)
+};
+
+type ChainFunds = Partial<{
+  [chain in SupportedMiddlewareChain]: {
+    [safeAddress: Address]: TokenAmountMap;
+  };
+}>;
+
+/**
+ * Fund an agent by sending funds to its service safe.
+ */
+const fundAgent = async ({
+  funds,
+  serviceConfigId,
+}: {
+  funds: ChainFunds;
+  serviceConfigId: ServiceConfigId;
+}): Promise<{ error: string | null }> =>
+  new Promise((resolve, reject) =>
+    fetch(`${BACKEND_URL_V2}/service/${serviceConfigId}/fund`, {
+      method: 'POST',
+      body: JSON.stringify(funds),
+      headers: { ...CONTENT_TYPE_JSON_UTF8 },
+    }).then((response) => {
+      if (response.ok) {
+        resolve(response.json());
+      } else {
+        reject('Failed to fund agent');
+      }
+    }),
+  );
+
+const useConfirmTransfer = () => {
+  const { selectedService } = useServices();
+  const { isPending, isSuccess, isError, mutateAsync } = useMutation({
+    mutationFn: async (funds: ChainFunds) => {
+      if (!selectedService) throw new Error('No service selected');
+
+      const serviceConfigId = selectedService.service_config_id;
+      await fundAgent({ funds, serviceConfigId });
+    },
+  });
+
+  const onFundAgent = useCallback(
+    async (fundsToPass: ChainFunds) => {
+      try {
+        await mutateAsync(fundsToPass);
+      } catch (error) {
+        console.error(error);
+      }
+    },
+    [mutateAsync],
+  );
+
+  return { onFundAgent, isLoading: isPending, isSuccess, isError };
+};
+
+type ConfirmTransferProps = {
+  fundsToTransfer: Record<string, number>;
+};
+
+export const ConfirmTransfer = ({ fundsToTransfer }: ConfirmTransferProps) => {
+  const { selectedAgentConfig, selectedService } = useServices();
+  const { serviceSafes } = useService(selectedService?.service_config_id);
+  const { onFundAgent, isLoading, isSuccess, isError } = useConfirmTransfer();
+
+  const [isTransferStateModalVisible, setIsTransferStateModalVisible] =
+    useState(false);
+
+  const { middlewareHomeChainId, evmHomeChainId } = selectedAgentConfig;
+  const serviceSafe = useMemo(
+    () => serviceSafes?.find(({ evmChainId }) => evmChainId === evmHomeChainId),
+    [serviceSafes, evmHomeChainId],
+  );
```

Comment:
should we move this as a function inside the `useService` hook? looks like something that could be reused